app/Http/Controllers/PostController.php

```
@@ -276,42 +358,45 @@ public function getPostStats($postId)
         if (!$user) {
             return response()->json(['error' => 'Unauthorized'], 401);
         }
 
         // Use firstOrCreate to check and create in single operation
-        $existingLike = $post->likes()->where('user_id', $user->id)->first();
+        $existingLike = $interactionPost->likes()->where('user_id', $user->id)->first();
         
         if ($existingLike) {
             $existingLike->delete();
             $liked = false;
             $countChange = -1;
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$countChange'. (undefined)

(UnusedLocalVariable)